tasks/dotnettracer.py

```diff
@@ -0,0 +1,180 @@
+"""
+customaction namespaced tasks
+"""
+from __future__ import print_function
+
+import glob
+import os
+import shutil
+import sys
+
+from invoke import task
+from invoke.exceptions import Exit, ParseError, Failure, UnexpectedExit
+from .ssm import get_signing_cert, get_pfx_pass
+
+@task
+def build(ctx, vstudio_root=None, arch="All", major_version='7', debug=False):
+    """
+    Build the custom action library for the .NET Tracer
+    """
+
+    if sys.platform != 'win32':
+        print("DotNet Tracer build is only for Win32")
+        raise Exit(code=1)
+
+    print("arch is {}".format(arch))
+    this_dir = os.getcwd()
+    solution_dir = os.getcwd()
+    tracer_home = os.path.join(solution_dir, "tracer_home")
+    output_path = os.path.join(solution_dir, "output")
+
+    pfxfile = None
+    pfxpass = None
+    remove_pfx = True
+    cmd = ""
+    configuration = "Release"
+    if debug:
+        configuration = "Debug"
+
+    ctx.run("nuget restore {solution_dir}\\Datadog.Trace.Minimal.sln".format(solution_dir=solution_dir))
+
+    cmd = "msbuild {solution_dir}\\Datadog.Trace.proj /t:{target} /p:Platform={arch} /p:Configuration={config} /p:TracerHomeDirectory={tracer_home} /p:RunWixToolsOutOfProc=true /p:MsiOutputPath={output_path}"
+
+    ## compile all binaries
+    run_cmd = cmd.format(
+        solution_dir=solution_dir,
+        target="CreateHomeDirectory",
+        arch=arch,
+        config=configuration,
+        tracer_home=tracer_home,
+        output_path=output_path
+    )
+    ctx.run(run_cmd)
+
+    ## pull the signing cert and password
+    try:
+        if sys.platform == 'win32' and os.environ.get('SIGN_WINDOWS'):
+            # get certificate and password from ssm
+            pfxfile = get_signing_cert(ctx)
+            pfxpass = get_pfx_pass(ctx)
+        else:
+            remove_pfx = False
+
+        ## sign all dll files
+        for f in glob.iglob("{tracer_home}/**/datadog*.dll".format(tracer_home=tracer_home), recursive=True):
+            sign_binary(ctx, f, pfxfile, pfxpass)
+
+        ## build the msi installers
+        run_cmd = cmd.format(
+            solution_dir=solution_dir,
+            target="MsiOnly",
+            arch=arch,
+            config=configuration,
+            tracer_home=tracer_home,
+            output_path=output_path
+        )
+        ctx.run(run_cmd)
+
+        ## sign msi installers
+        for f in glob.iglob("{output_path}/**/*.msi".format(output_path=output_path), recursive=True):
+            sign_binary(ctx, f, pfxfile, pfxpass)
+
+        ## build the nuget packages
+        cmd = "dotnet pack --no-restore --output {output_path} --configuration {configuration} -p:Platform=AnyCPU {proj}"
```

Comment:
You could add a `--no-build` argument if you wanted to, to keep the MSI contents the same as the NuGet package: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-pack#options

Reply:
I tried that and it failed. I think `dotnet pack` looks for the contents in a different path than the msi does.

```
C:\Program Files\dotnet\sdk\3.1.403\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(198,5):
  error NU5026: The file 'C:\build\src\Datadog.Trace\bin\Release\net45\Datadog.Trace.dll' to be packed was not found on disk.
  [C:\build\src\Datadog.Trace\Datadog.Trace.csproj]
```

Reply:
I tweaked the `.proj` file some more and got this to work for `Datadog.Trace`. ``Datadog.Trace.OpenTracing` still needs `--no-restore` since we don't build it for the msi.